memory-game.js

```diff
@@ -550,6 +553,44 @@ H5P.MemoryGame = (function (EventDispatcher, $) {
     if (parameters.behaviour && parameters.behaviour.useGrid && cardsToUse.length) {
       self.on('resize', scaleGameSize);
     }
+
+    /**
+     * Used for contracts.
+     * Checks the maximum score for this task.
+     *
+     * @returns {Number} The maximum score.
+     */
+    self.getScore = function () {
+      return score;
+    };
+
+    /**
+     * Used for contracts.
+     * Checks the maximum score for this task.
+     *
+     * @returns {Number} The maximum score.
+     */
+    self.getMaxScore = function () {
+      return 1;
+    };
+
+    /**
+     * getXAPIData
+     * Contract used by report rendering engine.
+     *
+     * @see contract at {@link https://h5p.org/documentation/developers/contracts#guides-header-6}
+     *
+     * @returns {Object} xAPI data
+     */
+    self.getXAPIData = function () {
+      var xAPIEvent = this.createXAPIEventTemplate('answered');
+      xAPIEvent.data.statement.result = {
```

Comment:
@ravimajithia This will work, but (close to?) all H5P content report the [full range of xAPI result properties](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#245-result) except `extensions`. Also, see https://h5p.org/documentation/developers/contracts#guides-header-6.

The xAPIEvent class provides you with the [`setScoredResult` function for that purpose](https://github.com/h5p/h5p-php-library/blob/master/js/h5p-x-api-event.js#L25). One thing that you'd normally want to add in addition is
```
if (this.isRoot()) {
  this.setActivityStarted();
}
```
in order to set the start time (commonly when the DOM is attached). Content types that inherit from H5P.Question get this "for free", but Memory Game would need to take care of it on its own. However, it doesn't :-D The timer can be paused, so currently it already overwrites whatever the `setScoredResult` function might have set for the duration, see https://github.com/h5p/h5p-memory-game/blob/master/memory-game.js#L124.

Also, you will by now have noticed that there's already an event created for the same purpose as the one that you added (https://github.com/h5p/h5p-memory-game/blob/master/memory-game.js#L121-L125). It uses the verb `completed` instead of `answered`, because that better reflects the user's experience: He/she _completed_ the game, but didn't answer anything. I know that (https://h5p.org/documentation/developers/contracts#guides-header-6) says "statement: Should contain the xAPI statement that is usually sent on 'answered.", but we're not in a "normal" _question type_ situation as Memory Game is not a question-like content type.

My suggestion would be to turn https://github.com/h5p/h5p-memory-game/blob/master/memory-game.js#L121-L124  into your `getXAPIData` function that returns the statement, pimp it with your `getScore` and `getMaxScore` functions and then replace https://github.com/h5p/h5p-memory-game/blob/master/memory-game.js#L122-L125 with a call to `getXAPIData` to get the statement, wrap it in a new xAPIEvent and trigger that.

Reply:
@otacke Thanks for giving me the background info. Regarding your suggestion, I would ask @fnoks for confirmation.

Reply:
@otacke I have updated the code according to your suggestion. Please feel free to review and let me know if you think it's worth mentioning.